components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayJMSMessageListener.java

```diff
@@ -119,26 +123,44 @@ private void handleNotificationMessage(String eventType, long timestamp, String
             gatewayEvent.getGatewayLabels().retainAll(gatewayArtifactSynchronizerProperties.getGatewayLabels());
             if (!gatewayEvent.getGatewayLabels().isEmpty()) {
                 String gatewayLabel = gatewayEvent.getGatewayLabels().iterator().next();
+                String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
```

Comment:
JMS message listener only works in super tenant Space, Therefore this was redundant.

Reply:
Fixed with c0ca5188d442023c03aae8c8a6862db89df93149

Reply:
this need to start tenant flow to the tenant which event belongs to